### DISCUSSION TITLE
rancher desktop doesn't launch

### DISCUSSION THREAD

**Discussion by ma740988** [rancher-sandbox/rancher-desktop#5291] [Q&A]
Rancher Desktop v1.9.1.  OS Windows Enterprise v10.   Device ID does not contain a dash. 
Issue:  Application fails to launch.   When I double click the Rancher Desktop icon, a wheel spins for a few seconds and the app doesn't launch. 

===== The log file. 
2023-08-03T18:40:01.182Z: Stopping background process Win32 socket proxy (pid <none>).
2023-08-03T18:40:05.765Z: Error starting up: c [Error]: C:\WINDOWS\system32\wsl.exe exited with code 1
    at ChildProcess.<anonymous> (c:\program files\rancher desktop\resources\app.asar\dist\app\background.js:1:15549)
    at ChildProcess.emit (node:events:527:28)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12) {
  command: [
    'C:\\WINDOWS\\system32\\wsl.exe',
    '--distribution',
    'wsl-vpnkit',
    '--exec',
    '/bin/wslpath',
    '-a',
    '-u',
    'c:\\program files\\rancher desktop\\resources\\resources\\linux\\bin\\docker-compose'
  ],
  stdout: '',
  code: 1,
  [Symbol(child-process.command)]: 'C:\\WINDOWS\\system32\\wsl.exe --distribution wsl-vpnkit --exec /bin/wslpath -a -u c:\\program files\\rancher desktop\\resources\\resources\\linux\\bin\\docker-compose'
}

====  End result when i try to run the wsl command from the command line.     
PS C:\Windows\System32\WindowsPowerShell\v1.0> C:\\WINDOWS\\system32\\wsl.exe --distribution wsl-vpnkit --exec /bin/wslpath -a -u "c:\\program files\\rancher desktop\\resources\\resources\\linux\\bin\\docker-compose"
/bin/wslpath: c:\\program files\\rancher desktop\\resources\\resources\\linux\\bin\\docker-compose
PS C:\Windows\System32\WindowsPowerShell\v1.0>

Thoughts?

**Comment by adamkpickering** [comment]
It's tough to say what is going on without more info (i.e. full logs with debug mode enabled). Can you provide logs?

That being said, we typically we see problems with systems that have the following things configured/installed:
- antivirus software
- a proxy
- a VPN

Do any of these describe your environment?

**Comment by jandubois** [comment]
This seems to be #4560. On startup Rancher Desktop tries to remove WSL integration from all distros which have not been configured for it (in case things got out of sync via e.g. a factory reset).

On some distros this code throws an error which is not handled. There is no known workaround until this bug is fixed.

**Comment by FocusedWanderer** [comment]
I was having this same issue: exit with error code 1 when calling an executable with wsl-vpnkit as distribution.

I finally opened up wsl-vpnkit directly in terminal and found C:\ was not mounted in /mnt/c & Rancher Desktop bin folder can't be reached.

looking at /etc/wsl.conf, I see automount is not enabled. I changed enabled to = true, exiting wsl-vpnkit, validated it fully stopped from powershell with wsl.exe -l -v, then tried to launch Rancher Desktop again.


```
WIN10VM:/mnt# cat /etc/wsl.conf
[boot]
systemd=false

[automount]
enabled=true

[network]
generateHosts=false
generateResolvConf=true

[interop]
enabled=true
appendWindowsPath=false
WIN10VM:/mnt#
```

I am unclear how this happened in wsl-vpnkit, it certainly worked at some point and then stopped. 

For my case specifically, my assumption is that some process changed automount defaults like Rancher Desktop or a corporate policy enforcement, or maybe most recent wsl-vpnkit ships this way..

hope this helps someone else - my dev environment was down for 3 days while I figured this out!
